hack/fork_exec_snoop.bt

```diff
@@ -0,0 +1,172 @@
+// PPID: This field stands for the Parent Process ID.
+// TGID: This field indicates the Thread Group ID.
+// PTGID: This field denotes the Parent Thread Group ID.
+// CMD: This field represents the command name (executable name) and args.
+//
+// Limitations: The output contains entries from all processes using fork/exec.
+
+BEGIN {
+    printf("%-25s %-12s %-12s %-8s %-8s %-8s %-8s %s\n",
+        "TRACEPOINT",
+        "T",
+        "DT",
+        "PID",
+        "PPID",
+        "TGID",
+        "PTGID",
+        "CMD"
+    );
+
+    @time_divisor = (uint64) 1000;
+}
+
+tracepoint:syscalls:sys_enter_clone
+{
+    $time = nsecs;
+    @time_start_fork_parent[tid] = $time;
+    @time_start_fork_child[tid] = $time;
+
+    if (@time_normalized)
+    {
+        printf("%-25s %-12u %-12s %-8d %-8d %-8d %-8d %s\n",
+            "syscalls:sys_enter_clone",
+            ($time - @time_normalized) / @time_divisor,
+            "",
+            curtask->pid,
+            curtask->parent->pid,
+            curtask->tgid,
+            curtask->parent->tgid,
+            comm
+        );
+    }
+}
+
+tracepoint:syscalls:sys_exit_clone
+/args->ret == 0 && @time_start_fork_child[curtask->parent->pid]/
+{
+    $time = nsecs;
+    $ptid = curtask->parent->pid;
+
+    if (@time_normalized)
+    {
+        printf("%-25s %-12u %-12u %-8d %-8d %-8d %-8d %s\n",
+            "syscalls:sys_exit_clone",
+            ($time - @time_normalized) / @time_divisor,
+            ($time - @time_start_fork_child[$ptid]) / @time_divisor,
+            curtask->pid,
+            curtask->parent->pid,
+            curtask->tgid,
+            $ptid,
+            comm
+        );
+    }
+
+    delete(@time_start_fork_child[$ptid]);
+}
+
+tracepoint:syscalls:sys_exit_clone
+/args->ret > 0 && @time_start_fork_parent[tid]/
+{
+    $time = nsecs;
+
+    if (@time_normalized)
+    {
+        printf("%-25s %-12u %-12u %-8d %-8d %-8d %-8d %s\n",
+            "syscalls:sys_exit_clone",
+            ($time - @time_normalized) / @time_divisor,
+            ($time - @time_start_fork_parent[tid]) / @time_divisor,
+            curtask->pid,
+            curtask->parent->pid,
+            curtask->tgid,
+            curtask->parent->tgid,
+            comm
+        );
+    }
+
+    delete(@time_start_fork_parent[tid]);
+}
+
+tracepoint:syscalls:sys_enter_exec*
+{
+    $time = nsecs;
+    @time_start_exec[tid] = $time;
+
+    if (@time_normalized)
+    {
+        printf("%-25s %-12u %-12s %-8d %-8d %-8d %-8d %s",
+            "syscalls:sys_enter_exec*",
+            ($time - @time_normalized) / @time_divisor,
+            "",
+            curtask->pid,
+            curtask->parent->pid,
+            curtask->tgid,
+            curtask->parent->tgid,
+            ""
+        );
+        join(args->argv);
+    }
+}
+
+tracepoint:syscalls:sys_exit_exec*
+/@time_start_exec[pid]/
+{
+    $time = nsecs;
+
+    if (!@time_normalized && strcontains(comm, "podman"))
```

Comment:
Could we do a filtering similar to https://github.com/containers/podman/blob/main/hack/podmansnoop#L126-L140 here?  We're waiting for the first "podman" process but then print everything for all processes after, also non-podman/non-containers related ones.

Reply:
Thanks for the explanation!  I was mostly curious.